### DISCUSSION TITLE
Verfügbare XMLRPC-Interfaces pro CCU/Homegear

### DISCUSSION THREAD

**Discussion by SukramJ** [SukramJ/aiohomematic#67] [Q&A] [answered]
Mit welchen XMLRPC-Interfaces  muss man bei einer CCU/Homegear rechnen?
Hintergrund: Ich würde eine entsprechende Auswahl gerne im ConfigFlow anbieten.

|Interface Name|Port|Port-TLS|PATH|
|---|---|---|---
|Wired|2000|42000|-|
|Bidcos-RF|2001|42001|-|
|HmIP-RF|2010|42010|-|
|Groups|9292|49292|/groups|

CUXd wird nicht unterstützt, weil hierfür BinRPC benötigt wird.

Fragen:
- Was ist mit RegaHSS Port(1999/41999) und RegaScript(8181/48181)? Hiermit kann ich nichts anfangen.
- Läuft HmIP-Wired über einen eigenen Port, oder über 2010?

**Comment by danielperna84** [comment, marked as answer]
CCU / Debmatic usw.: 2000, 2001, 2010, 9292  
Homegear: [2001, 2002, 2003](https://doc.homegear.eu/homegear/first_steps.html#how-to-interact-with-homegear) (wobei da vermutlich ein Schreibfehler bzgl. 2002 und 2003 im Link ist. Vermutlich ist bei einem von beidem SSL oder Auth deaktiviert).

Wir brauchen aber auf jeden Fall auch custom Ports. Es gab / gibt mindestens einen User dessen Setup mit NAT usw. so aussah, dass er 2 CCUs über eine IP angesprochen hat, aber die XML-RPCs auf verschiedene Ports gelegt hat (er hat auch einleuchtend erklärt, weshalb das bei ihm nicht anders geht, weiß es aber nicht mehr).

Cuxd ging ja schon vorher nicht. Dabei bleibt's auch. Braucht man ja eigentlich auch kaum wenn man Home Assistant benutzt und die Geräte dann stattdessen dort hat.

Zu den Fragen:
1. Ich habe es nie wirklich benutzt. Aber damit kann man auch manche Sachen auf einer CCU steuern. Habe es damals der Vollständigkeit halber mal zu den Konstanten hinzugefügt falls man das mal irgendwie einbauen will. Ich glaube damit kann man sich dann z.B. Räume / Gewerke holen. Aber das geht glaube ich auch über JSON-RPC. Also wirklich brauchen tut man's nicht. Und geht nur mit der CCU usw..
2. Läuft auch über 2010.

**Reply by SukramJ** [reply]
Erstmal vielen Dank für die Beantwortung der Fragen.
Es geht mit nicht um Einschränkungen, sondern um eine sinnvolle Vorbelegung der UI.
Wenn ich das richtig sehe können maximal 4 Schnittstellen vergeben werden.
Abhängig von der vorherigen Konfiguration ob TLS ja/nein werden die Ports vorbelegt.
Die Umsetzung im ConfigFlow ist leider sehr limitiert, aber ein paar Sachen gehen schon.
Ich hab schon mal  Dropdowns im CF gesehen, wodurch die Konfiguration etwas besser absichert werden könnte.
Problem ist das der CF bzw. die Konfiguration nachträglich nicht editiert werden kann.

**Reply by danielperna84** [reply]
Bei den Einschränkungen gibt's vermutlich auch keinen Weg zwischen einem _easy_ und _expert_ setup zu unterscheiden, oder? Denn theoretisch sollte man eine CCU auch via UPNP automatisch erkennen können (`http://1.2.3.4/upnp/basic_dev.cgi` auf einer CCU im Browser aufrufen). Beim _easy_ Modus würden dann einfach alle Ports der CCU genommen werden, beim _expert_ würde man sagen welche man will und mit welchem Port (und spart sich ggf. 1-2 Proxies wenn man eh nur HmIP benutzt, und hat völlige Flexibilität).

Homegear hat wohl auch UPNP, aber da weiß ich nicht was man da sehen würde. Sollte aber denke ich schon von einer CCU unterscheidbar sein. Wenn das so ist, könnte man da ja andere Ports anzeigen (sofern das im CF bekannt ist was per UPNP gefunden wurde).

Ist halt blöd bei den ganzen Variationen die es hier geben kann das alles unter einen Hut zu bringen. Und wenn man das ohne UPNP macht wäre es ja dann auch wieder alles manuell. Hmm, vielleicht hilft uns hier auch etwas die Meinung anderer weiter. Mir persönlich ist ein _manueller_ Weg bei dem der User alles von Hand (nach Doku) macht irgendwie lieber, denn so ein bisschen soll er sich ja schon mit der Materie auseinandersetzen. Und gerade mit dem CF könnte die Doku ja auch um einiges reduziert werden weil sich viele Fragen einfach nicht mehr stellen. Da sollten also selbst Laien die richtigen Entscheidungen treffen können finde ich.

**Reply by SukramJ** [reply]
Ich würde es im ersten Schritt auch bei der  manuellen Konfiguration belassen. Was soll den sonst noch in eine Roadmap :wink:.
Der jetztige CF funktioniert, aber ist noch recht komplex. 
Ich hätte gerne  einen Dropdown:
Auswahl HmIP-RF -> Vorbelegung Port 2010/42010.   
Auswahl BidcosRF -> Vorbelegung Port 2010/42010.   
Auswahl Groups -> Vorbelegung Port 9292/49292 -> Vorbelegung path mit /groups (ganz wichtig)  

Ziel sollte eine ConfigFlow sein der nicht erst durch eine umfangreiche Dokumentation erklärt werden muss.
Gefühlt gibt es die meisten Fragen bei pyhomematic zur initialen Konfiguration.

**Reply by SukramJ** [reply]
Funktioniert die URL (http://1.2.3.4/upnp/basic_dev.cgi) bei dir?

**Reply by danielperna84** [reply]
Sicher nur ein Tippfehler, aber bei BidcosRF wären es 2001/42001. Und bei Wired wäre ja noch 2000. Wired und RF sind nur bei IP zusammen. Dementsprechend sollte das Wired bei deinem IP-Eintrag dann auch dabei sein. Aber vermutlich auch nur ein Flüchtigkeitsfehler weil du's ja oben auch schon anders geschrieben hast. 👍

Ja, das ohne Doku machen zu können wäre schön. Da ist diese Vorauswahl dann schon sinnvoll. Da würde ich dann auch keine Extrawurst für Homegear machen. Da ist's einfach auch 2001, und wenn da jemand die anderen Ports benutzt, dann wird er das eh wissen. Solange man das Feld manuell beschreiben kann passt das also denke ich.

Und ja, bei mir geht der Link. Ich bekomme da so eine XML-Antwort. Ist in dem Fall Debmatic auf meinem ehemaligen HA-Pi. Aber ich weiß, dass ich das damals auch mit meiner CCU2 und später RaspberryMatic mal getestet habe und auch sowas bekommen habe. Kann natürlich sein, dass man das irgendwo in den Einstellungen der CCU ein- und ausschalten kann.
<img width="816" alt="Bildschirmfoto 2021-12-07 um 20 56 11" src="https://user-images.githubusercontent.com/7396998/145098482-9d1d5024-c3cd-4666-a403-0c82f131d97c.png">
:

**Reply by SukramJ** [reply]
Ja, das sind alles leider Flüchtigkeitsfehler. Die Tabelle oben zählt, und die Werte sollen änderbar sein.
UPNP lass ich erstmal.

**Reply by SukramJ** [reply]
Moin, 
ist der `path` eigentlich eine Konfigurationsoption für alle Schnittstellen, der nur bei `groups`?

**Reply by danielperna84** [reply]
Gibt's eigentlich nur bei den Gruppen soweit ich weiß.

**Reply by SukramJ** [reply]
Danke, dann nehme ich die Option aus dem ConfigFlow raus.

**Reply by danielperna84** [reply]
Und dann hardcoden für den Gruppen-Port? Find' ich grundsätzlich gut. Aber könnte sein, dass das für jemanden mit nicht-default-Ports zum Problem wird. Für diesen Sonderfall müsste dann noch eine Lösung gefunden werden.

**Reply by SukramJ** [reply]
Das ist die neue zweite Seite, nachdem auf der ersten Seite TLS ausgewählt wurde. Die Werte sind natürlich änderbar.

![Bild 13 12 21 um 10 52](https://user-images.githubusercontent.com/5650106/145790230-120de6d0-d25b-4122-9b18-33807456480f.jpg)

**Comment by Bimboboy007** [comment]
Hallo, hatte vorher die alte Integration über yaml da konnte ich mir eine Batterie Karte erstellen mit Volt anzeige der Batteriegeräten.
Hab es mir ihrer Integration versucht bekomme aber nur die üblichen Werte wie low Batt.
Wäre das ein großer aufwand dies zu bewerkstelligen ? Oder liegt es mir weil ich das nicht finden kann ?
Ansonsten ein großes Lob für ihre Arbeit.
Gruß
Boy Stefan

**Reply by stuckis6er** [reply]
Hallo,
wenn du das Gerät öffnest gibt es im Bereich Diagnose versteckte Entitäten. Die Entität Operating Voltage aktivieren und schon wird die Batteriespannung angezeigt.

Gruß Ralf

**Reply by Bimboboy007** [reply]
Danke hat funktioniert. Hab ich irgendwie übersehen. Homematic an sich ist nicht schlecht aber wenn es bei den Batterieangaben auch Prozent gäbe wäre auf jeden fall von Vorteil.

**Reply by SukramJ** [reply]
Bitte nicht irgendeine Diskussion verwenden. Der Inhalt sollte schon was mit dem Titel zu tun haben.
